idl/stm32h7-update.idol

```diff
@@ -69,5 +69,24 @@ Interface(
             ),
             idempotent: true,
         ),
+        "get_pending_boot_slot": (
```

Comment:
The idl/lpc55-update-server.idol `switch_default_image` API doesn't fit well enough to reuse here.
Someday we could unify the boot policy APIs.